Stomp-Client: add authentication to connection process

wherget opened the pull request:
This should fix #15

Known issues: you can't have passwords or usernames with colons in them. (Anyone know of suitable framework code to parse a URI.getUserInfo() in a standards-compliant way?)

wherget committed:
Stomp-Client: add authentication to connection process

swistaczek commented:
:+1:

nubbel commented on stomp-client/src/main/java/org/projectodd/stilts/stomp/client/ClientContextImpl.java:
```diff
@@ -81,6 +81,21 @@ public SSLContext getSSLContext() {
+
+    @Override
+    public String getPassword() {
+        return client.getAuthentiactionPassword();
```
There's a typo, should be `client.getAuthenticationPassword()`.

wherget committed:
fix typo authenti{ac=>ca}tionPassword

wherget commented:
An Idea I had to get around the colon-separation issue is to require username and password to be url-encoded and throw an error if there's more than one colon in the getUserInfo() string. What do you think?

wherget committed:
Rewrite extraction of authorization info from URL.

This should now be more compliant to RFC3986.
RFC3986 (3.2.1) states that the username ends at the first colon,
and everything after that is to be considered parameters for the
authorization scheme. (Here, it will be the password only.)

wherget commented:
I had a good re-read of RFC3986 (URI syntax) and rewrote the Authorization extraction to reflect that.
URL en/decoding ist done by the URI class already, so if you have no remaining issues, I'd be glad if you could merge those changes for the next version.
